src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -49,6 +49,20 @@ namespace drt {
 
 using utl::ThreadException;
 
+// TODO there should be a better way to get this info by getting the master
+// terms from OpenDB
+bool FlexPA::isStdCell(frInst* inst)
+{
+  return inst->getMaster()->getMasterType().isCore();
+}
+
+bool FlexPA::isMacroCell(frInst* inst)
+{
+  dbMasterType masterType = inst->getMaster()->getMasterType();
+  return (masterType.isBlock() || masterType.isPad()
+          || masterType == dbMasterType::RING);
```

Comment:
The only comment I have is this `dbMasterType::RING` inside a function called `isMacroCell`. It makes sense in the logic of DRT code, but I'm not sure how accurate is that in terms of the database. @maliberty any thoughts on this?

Reply:
From the manual:
```
Large macro that has an internal power mesh, and only exposes powerpin shapes that form a ring along the macro boundary
```
so a RING has no signal pins that could matter to drt/pa.    I doubt it matters whether is included or not but it is more like a macro than a stdcell.

Reply:
Thanks for the clarification! I think it's fine to keep it for now. The rest of the changes looks good too.